x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/job/persistence/JobConfigProvider.java

```diff
@@ -631,6 +631,44 @@ public void expandGroupIds(List<String> groupIds,  ActionListener<Set<String>> l
                 , client::search);
     }
 
+    /**
+     * Check if a group exists, that is there exists a job that is a member of
+     * the group. If there are one or more jobs that define the group then
+     * the listener responds with true else an error.
+     *
+     * @param groupId The group Id
+     * @param listener Returns true or failure
+     */
+    public void groupExists(String groupId, ActionListener<Boolean> listener) {
+        BoolQueryBuilder boolQueryBuilder = new BoolQueryBuilder();
+        boolQueryBuilder.filter(new TermQueryBuilder(Job.JOB_TYPE.getPreferredName(), Job.ANOMALY_DETECTOR_JOB_TYPE));
+        boolQueryBuilder.filter(new TermsQueryBuilder(Job.GROUPS.getPreferredName(), groupId));
+
+        SearchSourceBuilder sourceBuilder = new SearchSourceBuilder()
+                .query(boolQueryBuilder);
+        sourceBuilder.fetchSource(false);
```

Comment:
I think you should make `.size(0)` on the search as we don't really care about any of the hits, just the total number of them.

Reply:
👍